### ISSUE TITLE
Improve the reporting of a 404 response status code

### ISSUE THREAD

**Issue by rgl** [shijl0925/python-sonarqube-api#28]
While trying to test whether a given SonarQube project already has alm settings, I'm currently doing:

```python
c = sonarqube.SonarEnterpriseClient(sonarqube_url=sonarqube_address, token=sonarqube_api_token)
binding = c.alm_settings.get_binding(sonarqube_project_key)
```

When there is no alm settings defined at the project, `get_binding` raises a generic `sonarqube.utils.exceptions.ClientError` with the message `Error in request. Possibly client error [404]: Project 'xxx' is not bound to any ALM`.

But there is no straightforward way to deal with this expected error without looking at the exception message for `Possibly client error [404]`.

Can we have a better way of reporting a 404?

For example, maybe with a class named `NotFoundClientError`?

Also, maybe add a `response_code` property to the `ClientError` class too?

**Comment by shijl0925** [comment]
Thank you for your advice.

**Comment by rgl** [comment]
Thank you!

For reference, this was implemented in https://github.com/shijl0925/python-sonarqube-api/commit/c89f78eda3806047f5888f2b3604759fe65b2820 and released in [1.2.4](https://github.com/shijl0925/python-sonarqube-api/releases/tag/1.2.4).
